src/gp/inc/GPMSA.h

```diff
@@ -288,78 +339,116 @@ class GPMSAFactory
+  typename ScopedPtr<V>::Type totalMaxs;
 
-  BoxSubset<V, M> * totalDomain;
+  typename ScopedPtr<BoxSubset<V, M> >::Type totalDomain;
 
   std::vector<const BaseVectorRV<V, M> *> priors;
 
   // The hyperpriors
-  UniformVectorRV<V, M> * m_emulatorMean;  // scalar
-  GammaVectorRV<V, M> * m_emulatorPrecision;  // (scalar) gamma(a, b) shape-rate
-  BetaVectorRV<V, M> * m_emulatorCorrelationStrength;  // (dim scenariosspace + dim parameterspace)
-  GammaVectorRV<V, M> * m_discrepancyPrecision;  // (scalar) shape-rate
-  BetaVectorRV<V, M> * m_discrepancyCorrelationStrength;  // (dim scenariospace)
-  GammaVectorRV<V, M> * m_emulatorDataPrecision;  // (scalar) shape-rate
-  ConcatenatedVectorRV<V, M> * m_totalPrior;  // prior for joint parameters and hyperparameters
-
-  V * m_emulatorPrecisionShapeVec;
-  V * m_emulatorPrecisionScaleVec;
-  V * m_emulatorCorrelationStrengthAlphaVec;
-  V * m_emulatorCorrelationStrengthBetaVec;
-  V * m_discrepancyPrecisionShapeVec;
-  V * m_discrepancyPrecisionScaleVec;
-  V * m_discrepancyCorrelationStrengthAlphaVec;
-  V * m_discrepancyCorrelationStrengthBetaVec;
-  V * m_emulatorDataPrecisionShapeVec;
-  V * m_emulatorDataPrecisionScaleVec;
+  typename ScopedPtr<UniformVectorRV<V, M> >::Type m_emulatorMean;  // scalar
+  typename ScopedPtr<GammaVectorRV<V, M> >::Type m_emulatorPrecision;  // (dim num_svd_terms) gamma(a, b) shape-rate
+  typename ScopedPtr<GammaVectorRV<V, M> >::Type m_observationalPrecision;  // scalar gamma(a, b) shape-rate
+  typename ScopedPtr<BetaVectorRV<V, M> >::Type m_emulatorCorrelationStrength;  // (dim scenariosspace + dim parameterspace)
+  typename ScopedPtr<GammaVectorRV<V, M> >::Type m_discrepancyPrecision;  // (scalar) shape-rate
+  typename ScopedPtr<BetaVectorRV<V, M> >::Type m_discrepancyCorrelationStrength;  // (dim scenariospace)
+  typename ScopedPtr<GammaVectorRV<V, M> >::Type m_emulatorDataPrecision;  // (scalar) shape-rate
+  typename ScopedPtr<ConcatenatedVectorRV<V, M> >::Type m_totalPrior;  // prior for joint parameters and hyperparameters
+
+  typename ScopedPtr<V>::Type m_emulatorPrecisionShapeVec;
+  typename ScopedPtr<V>::Type m_emulatorPrecisionScaleVec;
+  typename ScopedPtr<V>::Type m_observationalPrecisionShapeVec;
+  typename ScopedPtr<V>::Type m_observationalPrecisionScaleVec;
+  typename ScopedPtr<V>::Type m_emulatorCorrelationStrengthAlphaVec;
+  typename ScopedPtr<V>::Type m_emulatorCorrelationStrengthBetaVec;
+  typename ScopedPtr<V>::Type m_discrepancyPrecisionShapeVec;
+  typename ScopedPtr<V>::Type m_discrepancyPrecisionScaleVec;
+  typename ScopedPtr<V>::Type m_discrepancyCorrelationStrengthAlphaVec;
+  typename ScopedPtr<V>::Type m_discrepancyCorrelationStrengthBetaVec;
+  typename ScopedPtr<V>::Type m_emulatorDataPrecisionShapeVec;
+  typename ScopedPtr<V>::Type m_emulatorDataPrecisionScaleVec;
 
   // The gaussian process object to build
-  GPMSAEmulator<V, M> * gpmsaEmulator;
+  typename ScopedPtr<GPMSAEmulator<V, M> >::Type gpmsaEmulator;
```

Comment:
Ugh, thanks for doing all that.  I think when I initially implemented this we didn't have the `ScopedPtr` foo.